webcrypto/subtle_crypto.go

```diff
@@ -372,33 +361,13 @@ func (sc *SubtleCrypto) Verify(algorithm, key, signature, data goja.Value) *goja
 			return
 		}
 
-		switch normalizedAlgorithm.Name {
-		case HMAC:
-			keyAlgorithm, ok := ck.Algorithm.(hasHash)
-			if !ok {
-				reject(NewError(InvalidAccessError, "key algorithm does not describe a HMAC key"))
-				return
-			}
-
-			keyHandle, ok := ck.handle.([]byte)
-			if !ok {
-				reject(NewError(InvalidAccessError, "key handle is of incorrect type"))
-				return
-			}
-
-			hashFn, ok := getHashFn(keyAlgorithm.hash())
-			if !ok {
-				reject(NewError(NotSupportedError, "unsupported hash algorithm "+keyAlgorithm.hash()))
-				return
-			}
-
-			hasher := hmac.New(hashFn, keyHandle)
-			hasher.Write(signedData)
-
-			resolve(hmac.Equal(signatureData, hasher.Sum(nil)))
-		default:
-			reject(NewError(NotSupportedError, fmt.Sprintf("unsupported algorithm %q", normalizedAlgorithm.Name)))
+		verified, err := verifier.Verify(ck, signatureData, signedData)
+		if err != nil {
+			reject(err)
+			return
 		}
+
+		resolve(rt.ToValue(verified))
```

Comment:
```suggestion
		resolve(verified)
```

No need for this and also you shouldn't access the runtime off event loop